mock-api/msw/handlers.ts

```diff
@@ -1053,6 +1053,78 @@ export const handlers = makeHandlers({
 
     return { rules: R.sortBy(rules, (r) => r.name) }
   },
+  vpcRouterList({ query }) {
+    const vpc = lookup.vpc(query)
+    const routers = db.vpcRouters.filter((r) => r.vpc_id === vpc.id)
+    return paginated(query, routers)
+  },
+  vpcRouterCreate({ body, query }) {
+    const vpc = lookup.vpc(query)
+    errIfExists(db.vpcRouters, { vpc_id: vpc.id, name: body.name })
+
+    const newRouter: Json<Api.VpcRouter> = {
+      id: uuid(),
+      vpc_id: vpc.id,
+      kind: 'custom',
+      ...body,
+      ...getTimestamps(),
+    }
+    db.vpcRouters.push(newRouter)
+    return json(newRouter, { status: 201 })
+  },
+  vpcRouterView: ({ path, query }) => lookup.vpcRouter({ ...path, ...query }),
+  vpcRouterUpdate({ body, path, query }) {
+    const router = lookup.vpcRouter({ ...path, ...query })
+
+    if (body.name) {
+      router.name = body.name
+    }
+
+    if (typeof body.description === 'string') {
+      router.description = body.description
+    }
+
+    return router
+  },
+  vpcRouterDelete({ path, query }) {
+    const router = lookup.vpcRouter({ ...path, ...query })
+    db.vpcRouters = db.vpcRouters.filter((r) => r.id !== router.id)
+    return 204
+  },
+  vpcRouterRouteList: ({ query }) => {
+    const { project, router, vpc } = query
+    const vpcRouter = lookup.vpcRouter({ project, router, vpc })
+    const routes = db.vpcRouterRoutes.filter((r) => r.vpc_router_id === vpcRouter.id)
+    return paginated(query, routes)
+  },
+  vpcRouterRouteCreate({ body, query }) {
+    const vpcRouter = lookup.vpcRouter(query)
+    const newRoute: Json<Api.RouterRoute> = {
+      id: uuid(),
+      vpc_router_id: vpcRouter.id,
+      kind: 'custom',
+      ...body,
+      ...getTimestamps(),
+    }
+    db.vpcRouterRoutes.push(newRoute)
+    return json(newRoute, { status: 201 })
+  },
+  vpcRouterRouteView: ({ path, query }) => lookup.vpcRouterRoute({ ...path, ...query }),
+  vpcRouterRouteUpdate({ body, path, query }) {
+    const route = lookup.vpcRouterRoute({ ...path, ...query })
+    if (body.destination) {
+      route.destination = body.destination
+    }
+    if (body.target) {
+      route.target = body.target
+    }
+    return route
+  },
```

Comment:
This should handle name and description too. I looked through other update endpoints and I wish we had a less manual way of doing these, but it's fine.